include/kllvm/codegen/ProofEvent.h

```diff
@@ -228,6 +239,10 @@ class proof_event {
   [[nodiscard]] llvm::BasicBlock *pattern_matching_failure(
       kore_composite_pattern const &pattern, llvm::BasicBlock *current_block);
 
+  [[nodiscard]] llvm::BasicBlock *tail_call_info(
+      std::string const &caller_name, bool is_tail,
+      llvm::Instruction *insert_before, llvm::BasicBlock *current_block);
```

Comment:
you don't really need both insert_before and current_block, right? could we change this signature so that you pass one or the other?